chain/src/tests/block_assembler.rs

```diff
@@ -557,6 +557,19 @@ fn test_package_multi_best_scores() {
         TxEntry::dummy_resolve(tx3_1.clone(), 0, Capacity::shannons(1000), 1000),
         TxEntry::dummy_resolve(tx4_1.clone(), 0, Capacity::shannons(300), 250),
     ];
+
+    let shared_tip_number = shared.snapshot().tip_number().into();
```

Comment:
suggest to move this patch block between `plug_entry` and `package_txs`.

Reply:
If I move this patch block between `plug_entry` and `package_txs`, the error will occur:
```bash
    Finished test [unoptimized + debuginfo] target(s) in 0.22s
     Running unittests src/lib.rs (target/debug/deps/ckb_chain-26e466a4c3646cb7)

running 1 test
thread 'tests::block_assembler::test_package_multi_best_scores' panicked at 'assertion failed: `(left == right)`
  left: `[Byte32(0xcd0d80078c5fb0aed3cc059f73562cb2f1e4a71823f8d720f83ff9f2c9214769), Byte32(0x001d9a4774b77d4d7d4572a52e7f88b267da4b483b1430ee3b1dbe8a555c0ff5), Byte32(0x790691b80c0e6dfca591bf62f72b4b74018ab296d8fc12c1f078397e8e1e2a2b), Byte32(0xf3ace5260c0e0c38b70235466c75645d70ab41d052035d3a636ead24c8c5c935), Byte32(0x42afa370eeb4ef6707a090317bb07740c97d4762c20bfdf114665e7422489524), Byte32(0xa181a0f4a636b91d7efa55ee49b9a42787847242d76c14343cf25eb116d052c2), Byte32(0xe9d809a20bef9c45bdc1258dadb55e7db72b88029b292633e6d748af00f3b6b3), Byte32(0x8006255ad512edfaee5b658424265bef712f49f82d8e4aae072c144232e167d2), Byte32(0xf2bdff1e814a130cf3a76a7e6147ed9c0fd6b367d4be44f1113149cc932b2b0e)]`,
 right: `[Byte32(0xcd0d80078c5fb0aed3cc059f73562cb2f1e4a71823f8d720f83ff9f2c9214769), Byte32(0x001d9a4774b77d4d7d4572a52e7f88b267da4b483b1430ee3b1dbe8a555c0ff5), Byte32(0x790691b80c0e6dfca591bf62f72b4b74018ab296d8fc12c1f078397e8e1e2a2b), Byte32(0xf3ace5260c0e0c38b70235466c75645d70ab41d052035d3a636ead24c8c5c935), Byte32(0x42afa370eeb4ef6707a090317bb07740c97d4762c20bfdf114665e7422489524), Byte32(0xa181a0f4a636b91d7efa55ee49b9a42787847242d76c14343cf25eb116d052c2), Byte32(0xe9d809a20bef9c45bdc1258dadb55e7db72b88029b292633e6d748af00f3b6b3), Byte32(0x8006255ad512edfaee5b658424265bef712f49f82d8e4aae072c144232e167d2), Byte32(0xd551fbc7a1b0f6ef4e1135ce320a562f66b171a1973ba4800e38810e5c618d9d), Byte32(0xf2bdff1e814a130cf3a76a7e6147ed9c0fd6b367d4be44f1113149cc932b2b0e)]`: best scored txs', chain/src/tests/block_assembler.rs:389:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
test tests::block_assembler::test_package_multi_best_scores ... FAILED

failures:

failures:
    tests::block_assembler::test_package_multi_best_scores

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 45 filtered out; finished in 0.41s

error: test failed, to rerun pass `-p ckb-chain --lib`

```